src/check.js

```diff
@@ -48,61 +47,54 @@ function discoverPropertyDep(rootDir, deps, property, depName) {
   return lodash.intersection(deps, propertyDeps);
 }
 
-function getDependencies(dir, filename, deps, parser, detectors) {
-  return new Promise((resolve, reject) => {
-    getContent(filename)
-      .then((content) => resolve(parser(content, filename, deps, dir)))
-      .catch((error) => reject(error));
-  }).then((ast) => {
-    // when parser returns string array, skip detector step and treat them as dependencies.
-    const dependencies =
-      lodash.isArray(ast) && ast.every(lodash.isString)
-        ? ast
-        : lodash(getNodes(ast))
-            .map((node) => detect(detectors, node, deps))
-            .flatten()
-            .uniq()
-            .map(requirePackageName)
-            .thru((_dependencies) =>
-              parser === availableParsers.typescript
-                ? // If this is a typescript file, importing foo would also use @types/foo, but
-                  // only if @types/foo is already a specified dependency.
-                  lodash(_dependencies)
-                    .map((dependency) => {
-                      const atTypesName = getAtTypesName(dependency);
-                      return deps.includes(atTypesName)
-                        ? [dependency, atTypesName]
-                        : [dependency];
-                    })
-                    .flatten()
-                    .value()
-                : _dependencies,
-            )
-            .value();
+async function getDependencies(dir, filename, deps, parser, detectors) {
+  const result = await parser(filename, deps, dir);
```

Comment:
We now get the result directly from the parser/special here, async.

Except that, not much changed, it's just the indentation.